speech/speech_samples.rb

```diff
@@ -12,160 +12,168 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def speech_sync_recognize project_id: nil, audio_file_path: nil
+def speech_sync_recognize audio_file_path: nil
 # [START speech_sync_recognize]
-  # project_id      = "Your Google Cloud project ID"
   # audio_file_path = "Path to file on which to perform speech recognition"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize
+  audio_file = File.binread audio_file_path
+  config     = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio      = { content: audio_file }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
```

Comment:
# Ruby Idiom Issue 🛑

Reply:
With Ruby, when methods accept a hash... Rubyists pass the hash parameter in a way such that it LOOKS like normal named parameters. Just by reading the code, there actually is no way to tell if the method takes a Hash or regular named params

Reply:
SOLUTION: Instead of `recognize` taking 2 params, it should take 1

That way, people can...

```ruby
recognize content: audio_file, encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US"
```

Currently, you have to do this ☹️ 

```ruby
recognize({ content: audio_file }, encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US")
```

NOTE: parens **required** ☹️

Reply:
**IDEAL** Can we have `content` be the first positional argument again? 😄

Reply:
## ❗️ ACTUALLY.... 

`content:` or `uri:` don't fit in very well with `encoding:` `sample_rate_hertz:` etc...

And Video uses different `input_content:` and `input_uri:`

Could the named params be consistent???

I'm OK with the named params being all grouped together ... but let's make them consistent ...

### Let's put a new method on top of this and handle the params ourselves... `**kwargs` and handle input_uri and input_content

Reply:
```ruby
def recognize **kwargs
  audio = {}
  if kwargs[:audio_uri]
    audio[:uri] = kwargs.delete :audio_uri
  elsif
    audio[:content] = kwargs.delete :audio_content
  else
    raise "No dawg, u needs audio yo! 🐕"
  end

  super audio, kwargs
end
```

Reply:
#### *Consistency please too* 😄